src/fswAlgorithms/attGuidance/prescribedTrans/_UnitTest/test_ prescribedTrans.py

```diff
@@ -0,0 +1,187 @@
+
+    # Create test thread
+    testProcessRate = macros.sec2nano(0.1)     # update process rate update time
+    testProc = unitTestSim.CreateNewProcess(unitProcessName)
+    testProc.addTask(unitTestSim.CreateNewTask(unitTaskName, testProcessRate))
+
+    # Construct algorithm and associated C++ container
+    PrescribedTransConfig = prescribedTrans.PrescribedTransConfig()
+    PrescribedTransWrap = unitTestSim.setModelDataWrap(PrescribedTransConfig)
+    PrescribedTransWrap.ModelTag = "prescribedTrans"                                 # update python name of test module
+
+    # Add test module to runtime call list
+    unitTestSim.AddModelToTask(unitTaskName, PrescribedTransWrap, PrescribedTransConfig)
+
+    # Initialize the test module configuration data
+    transAxis_M = np.array([0.5, 0.0, 0.5 * np.sqrt(3)])
+    PrescribedTransConfig.transAxis_M = transAxis_M
+    PrescribedTransConfig.scalarAccelMax = scalarAccelMax  # [rad/s^2]
+    PrescribedTransConfig.r_FM_M = scalarPosInit * transAxis_M
+    PrescribedTransConfig.rPrime_FM_M = np.array([0.0, 0.0, 0.0])
+    PrescribedTransConfig.rPrimePrime_FM_M = np.array([0.0, 0.0, 0.0])
+    PrescribedTransConfig.omega_FM_F = np.array([0.0, 0.0, 0.0])
+    PrescribedTransConfig.omegaPrime_FM_F = np.array([0.0, 0.0, 0.0])
+    PrescribedTransConfig.sigma_FM = np.array([0.0, 0.0, 0.0])
+
+    # Create input message
+    scalarVelRef = 0.0  # [m/s]
+
+    PrescribedTransMessageData = messaging.PrescribedTransMsgPayload()
+    PrescribedTransMessageData.scalarPos = scalarPosRef
+    PrescribedTransMessageData.scalarVel = scalarVelRef
+    PrescribedTransMessage = messaging.PrescribedTransMsg().write(PrescribedTransMessageData)
+    PrescribedTransConfig.prescribedTransInMsg.subscribeTo(PrescribedTransMessage)
+
+    # Setup logging on the test module output message so that we get all the writes to it
+    dataLog = PrescribedTransConfig.prescribedMotionOutMsg.recorder()
+    unitTestSim.AddModelToTask(unitTaskName, dataLog)
+
+    # Need to call the self-init and cross-init methods
+    unitTestSim.InitializeSimulation()
+
+    # Set the simulation time
+    simTime = np.sqrt(((0.5 * np.abs(scalarPosRef - scalarPosInit)) * 8) / scalarAccelMax) + 5
+    unitTestSim.ConfigureStopTime(macros.sec2nano(simTime))
+
+    # Begin the simulation time run set above
+    unitTestSim.ExecuteSimulation()
+
+    # Extract logged data
+    r_FM_M = dataLog.r_FM_M
+    rPrime_FM_M = dataLog.rPrime_FM_M
+    timespan = dataLog.times()
+
+    scalarVel_Final = np.linalg.norm(rPrime_FM_M[-1, :])
+    scalarPos_Final = np.linalg.norm(r_FM_M[-1, :])
+
+    # Plot r_FM_F
+    r_FM_M_Ref = scalarPosRef * transAxis_M
+    r_FM_M_1_Ref = np.ones(len(timespan)) * r_FM_M_Ref[0]
+    r_FM_M_2_Ref = np.ones(len(timespan)) * r_FM_M_Ref[1]
+    r_FM_M_3_Ref = np.ones(len(timespan)) * r_FM_M_Ref[2]
+
+    plt.figure()
+    plt.clf()
+    plt.plot(timespan * 1e-9, r_FM_M[:, 0], label='$r_{1}$')
```

Comment:
the `utilities.macros` package as a NANO2Sec conversion tool

Reply:
Done